doc/source/coregistration.md

```diff
@@ -16,250 +16,472 @@ kernelspec:
+- {func}`~xdem.coreg.AffineCoreg.to_matrix` to convert the transform to a 4x4 transformation matrix, if possible,
+- {func}`~xdem.coreg.AffineCoreg.from_matrix` to create a coregistration from a 4x4 transformation matrix.
+
+```{margin}
+<sup>1</sup>In a style inspired by [scikit-learn's pipelines](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html#sklearn-linear-model-linearregression).
+```
 
 **Inheritance diagram of implemented coregistrations:**
 
 ```{eval-rst}
-.. inheritance-diagram:: xdem.coreg.base xdem.coreg.affine xdem.coreg.biascorr
+.. inheritance-diagram:: xdem.coreg.base.Coreg xdem.coreg.affine xdem.coreg.biascorr
         :top-classes: xdem.coreg.Coreg
 ```
 
-See {ref}`biascorr` for more information on non-rigid transformations ("bias corrections").
+**See {ref}`biascorr`** for more information on non-rigid transformations.
+
+## Coregistration methods
 
-(coregistration-nuthkaab)=
+```{important}
+Below we **create misaligned elevation data to examplify the different methods** in relation to their type of affine transformation.
 
-## Nuth and Kääb (2011)
+See coregistration on real data in the **{ref}`examples-basic` and {ref}`examples-advanced` gallery examples**!
+```
+
+(nuthkaab)=
+### Nuth and Kääb (2011)
 
 {class}`xdem.coreg.NuthKaab`
 
-- **Performs:** translation and vertical shift.
-- **Supports weights** (soon)
-- **Recommended for:** Noisy data with low rotational differences.
+- **Performs:** Horizontal and vertical shifts.
+- **Supports weights:** Planned.
+- **Pros:** Refines sub-pixel horizontal shifts accurately, with fast convergence.
+- **Cons:** Diverges on flat terrain, as landforms are required to constrain the fit with aspect and slope.
 
-The Nuth and Kääb ([2011](https://doi.org/10.5194/tc-5-271-2011)) coregistration approach is named after the paper that first implemented it.
-It estimates translation iteratively by solving a cosine equation to model the direction at which the DEM is most likely offset.
-First, the DEMs are compared to get a dDEM, and slope/aspect maps are created from the reference DEM.
-Together, these three products contain the information about in which direction the offset is.
-A cosine function is solved using these products to find the most probable offset direction, and an appropriate horizontal shift is applied to fix it.
-This is an iterative process, and cosine functions with suggested shifts are applied in a loop, continuously refining the total offset.
-The loop stops either when the maximum iteration limit is reached, or when the NMAD between the two products stops improving significantly.
+The [Nuth and Kääb (2011)](https://doi.org/10.5194/tc-5-271-2011) coregistration approach estimates a horizontal
+translation iteratively by solving a cosine equation between the terrain slope, aspect and the elevation differences.
+The iteration stops if it reaches the maximum number of iteration limit or if the tolerance does not improve.
```

Comment:
it's not the tolerance that improves but the shift falls below a certain threshold.

Reply:
Nice catch! :sweat_smile: